src/main/java/assignment/LaunchInterceptorConditions.java

```diff
@@ -41,23 +41,109 @@ public static boolean ConditionThree(double AREA1, int[][] POINTS) {
     }
 
     public static boolean ConditionFour(int QPTS, int QUADS, int[][] POINTS) {
-        // do something
-        return true;
+        if(QPTS < 2 || QPTS > POINTS.length){
+            return false;
+        }
+    
+        if(QUADS < 1 || QUADS > 3){
+            return false;
+        }
+
+        for(int i = 0; i < POINTS.length - QPTS -1; i++){
+            boolean[] FoundQuads = {false, false, false, false};
+            for(int j = 0; j < QPTS; j++){
+                if(POINTS[i][0] >= 0 && POINTS[i][1] >= 0){
```

Comment:
It doesn't make sense to test the same point in each loop iteration. Probably `i` should be `i+j` instead.